test/server/TypeDBRunner.java

```diff
@@ -68,24 +70,40 @@ protected static String host() {
 
     protected abstract int port();
 
-    protected abstract void verifyPortUnused();
-
-    protected void verifyPortUnused(int port) {
-        if (isPortOpen(host(), port)) throw new RuntimeException(name() + ": unable to start. port " + port + " is used by another process.");
+    protected static List<Integer> findUnusedPorts(int count) {
+        assert count > 0;
+        try {
+            while (true) {
+                List<Integer> ports = new ArrayList<>(count);
+                // using port 0 automatically allocates a valid free port
+                ServerSocket seed = new ServerSocket(0);
+                ports.add(seed.getLocalPort());
+                seed.close();
+                for (int i = 1; i < count; i++) {
+                    if (isPortUnused(ports.get(0) + i)) {
+                        ports.add(ports.get(0) + i);
+                    } else {
+                        break;
+                    }
+                }
+                if (ports.size() == count) return ports;
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error while searching for unused port.");
+        }
```

Comment:
find unused server-side ports to boot the server with! 

Deficiency: keeps retrying forever...?

Reply:
Yeah, let's limit the loop such that it fails after say 15 failed attempts.